### DISCUSSION TITLE
Why no `exports` in Firebase @exp packaging?

### DISCUSSION THREAD

**Discussion by akauppi** [firebase/firebase-js-sdk#4532] [Q&A] [answered]
Is there a reason why the Firebase @exp `package.json`s don't include an `"exports"` field?

https://medium.com/swlh/npm-new-package-json-exports-field-1a7d1f489ccf

https://nodejs.org/api/packages.html#packages_exports

I use `"exports"` in the ES-only modules that I publish (well, just two) and have found it useful. 

Is Firebase going to be using it, has it been turned down?

I'm building the app using [Vite](https://vitejs.dev) or Rollup. Both of these are capable of handing the `"exports"` field correctly.

**Comment by Feiyang1** [comment, marked as answer]
We will support it, just haven't gotten around to it yet. I will make a PR soon. will also add the support for the existing SDK.

**Reply by jacob-8** [reply]
@Feiyang1 What's the timeline on adding the exports field to firebase@exp? This is the only thing stopping me from using modular Firebase on an SSR project. If it helps, this could be done first for the firebase@exp version and not wait on getting this into the main Firebase JS SDK as it looks you intend to: https://github.com/firebase/firebase-js-sdk/issues/3069

**Reply by Feiyang1** [reply]
We have published a beta release with `exports` support. You can install it by `npm install firebase@beta`.
